### ISSUE TITLE
Lightbox gallery doesn't load "fullsize" derivatives for TIFFs

### ISSUE THREAD

**Issue by allanaaa** [omeka/omeka-s#2068]
When TIFFs are uploaded to items and Lightbox gallery is enabled, it will fail to display on the item page, and TIFFs will be relegated to "other media". When a Lightbox appears on a media page, it will show "Failed to load content". 
![tiff1](https://github.com/omeka/omeka-s/assets/28012148/20621d51-b52a-4454-a4d0-6b188f00209b)
![tiff2](https://github.com/omeka/omeka-s/assets/28012148/cd7c0e97-1145-441e-b86f-3707448938c3)

Is there any way to get Lightbox to look for fullsize derivatives instead of the original TIFF? We have never warned against TIFFs explicitly before, so I think people assumed we could handle them gracefully. I'm sure users would like something like full-scan book pages to display nicely in the Lightbox.

**Comment by zerocrates** [comment]
Here's what the rules are:

You get rendered in there if the file is a bmp, gif, jpg, png, svg, pdf, or a video, otherwise not (the common thread being that browsers can natively display these).

On fullsize: the problem is just that we don't check for that currently, there's an assumption in the code that everything that gets sent as ok to the lightbox should always use its original. So we'd need to build in some manual checking for tiffs.

JPEG2000 is the other format that gets occasional use that I can think of that probably should be treated similarly.

**Comment by zerocrates** [comment]
There's two basic moves here:

- make the gallery show _anything_ with a thumbnail, or at least with a generated one
- carve out specific treatment for image/tiff and image/jp2 (maybe some others also but none as significant) to be treated this way

What we're aiming to avoid is things that really should get displayed with their native renderer but that also have a thumbnail from getting caught up in the light gallery and displaying a less-useful version of themselves. The big things here would be stuff like embeds and similar media that can have thumbnails but also a potentially complex render.

The safer route is going to be to specifically list out the stuff that should get this treatment, I think.

**Comment by zerocrates** [comment]
Up on a branch: `tiff-gallery`.

The direction I went here was: things will get included if we have thumbnails for them and they're any otherwise-unrecognized image/* mimetype. TIFF and JPEG2000 would be the most obvious, but others are possible.

**Comment by allanaaa** [comment]
Looks good on the item page (showing all resource blocks here): http://dev.omeka.org/amayer/amayer-s/omeka-s/s/my-omeka-s-site/item/10088#lg=1&slide=0

Media page, not yet: http://dev.omeka.org/amayer/amayer-s/omeka-s/s/my-omeka-s-site/media/10089#lg=1&slide=0

**Comment by zerocrates** [comment]
OK, I had missed the "media" aspect of this entirely but I see now that it's in your first post.

The media gallery has a slightly different problem: when you choose that block, we _always_ render the gallery with the media, even if its a media the gallery can't show. So you get the error you have above for tiffs, which we can fix, but if the media is something like IIIF or an oEmbed or something, it's just not going to work.

I think what I'll do to fix this is to make the media block for this do the same checking that the item one does, for whether the media is supported by the gallery, and if it's not supported, then we'll just render the standard render (so you'd get the IIIF zoomer for a IIIF image, not the gallery, even though you picked the gallery in your resource page block config). I think that's the best answer here, and doing this should automatically pick up tiffs (they'll get the gallery with their large thumb, as done with this branch currently for the item gallery).

**Comment by zerocrates** [comment]
OK latest pull on that branch should fix the media gallery to work like the item one for these filetypes.

**Comment by zerocrates** [comment]
you have a tiff and a jp2 on the same item and it's only showing one of them?

**Comment by allanaaa** [comment]
Yep: http://dev.omeka.org/amayer/amayer-s/omeka-s/s/my-omeka-s-site/item/10088#lg=1&slide=0

On each media page, the lightbox loads each fine - just the item page is only showing one.

**Comment by allanaaa** [comment]
I just switched the media order (and the primary media) and the same thing - only the second media is showing in the lightbox on the item page, though each are displaying fine everywhere else they're expected.

**Comment by allanaaa** [comment]
That's done! I can't see any other issues with this - other places call the derivatives successfully, etc. There's no way to drop a lightbox gallery in via a page block, right? If not, we're good to go.

**Comment by zerocrates** [comment]
I think we've covered everywhere it exists.
